srcs/main/prompt.c

```diff
@@ -6,27 +6,55 @@
 /*   By: jbarbay <jbarbay@student.42singapore.sg    +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2024/01/12 18:32:31 by jbarbay           #+#    #+#             */
-/*   Updated: 2024/01/15 22:38:57 by jbarbay          ###   ########.fr       */
+/*   Updated: 2024/01/21 13:32:58 by jbarbay          ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 // Read line
 // Exit when type exit
-// History
+// Add to history (except if the line is empty)
+// Create tokens based on the input
+// If tokens is NULL, it means there was an error (with message): new prompt
 
 #include "../../includes/minishell.h"
 
+int	is_exit(char *line)
+{
+	char	*trimmed;
+
+	trimmed = ft_strtrim(line, WHITESPACE);
+	if (ft_strncmp(trimmed, "exit", 5) == 0)
+	{
+		free(trimmed);
+		return (1);
+	}
+	free(trimmed);
+	return (0);
+}
+
 int main(void)
 {
 	char	*line;
+	t_token	*tokens;
 
```

Comment:
I will need init function (+ return if failure) so 2 lines here before the while loop so please consider it for your main part to do not > 25 lines
init(&data, envp))
while (1)
For my data structure its: 
struct s_data
{
	char						exit_flag;
	char						**envp;
+ TBD
};

Reply:
Okay, I reduced the functions to 25 lines, but for main, since we both need to integrate many things, we will refactor to 25 lines at the end when everything is added.